internal/providers/store.go

```diff
@@ -0,0 +1,147 @@
+// Copyright 2024 Stacklok, Inc
```

Comment:
Similar to my other PR - this relies on the existing tests in the controlplane to validate it. It should get its own dedicated test suite in due course.